File: binary-tree-level-order-traversal/binary-tree-level-order-traversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def levelOrder(self, root: TreeNode) -> List[List[int]]:
        if not root:
            return []
        
        res = []
        def order(p, d):
            if (len(res) < d):
                res.append([p.val])
            else:
                res[d-1].append(p.val)
                
            if p.left:
                order(p.left, d+1)
            if p.right:
                order(p.right, d+1)
        order(root, 1)
        return res

File: binary-tree-level-order-traversal/binary-tree-level-order-traversal.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number[][]}
 */
var levelOrder = function(root) {
    let res = []
    if (!root) {
        return res
    }
    res.push([root.val])
    let nextNodes = []
    if (root.left) {
        nextNodes.push(root.left)
    }
    if (root.right) {
        nextNodes.push(root.right)
    }
    
    while (nextNodes.length > 0) {
        let nodeStack = []
        let sameLevel = []
        for (let node of nextNodes) {
            sameLevel.push(node.val)
            if (node.left) {
                nodeStack.push(node.left)
            }
            if (node.right) {
                nodeStack.push(node.right)
            }
        }
        nextNodes = nodeStack
        res.push(sameLevel)
    }
    return res
};